src/lib/ibcclient.ts

```diff
@@ -41,13 +52,25 @@ function ibcRegistry(): Registry {
   ]);
 }
 
+function timestampFromDateNanos(date: ReadonlyDateWithNanoseconds): Timestamp {
+  const nanos = (date.getTime() % 1000) * 1000000 + (date.nanoseconds ?? 0);
+  return Timestamp.fromPartial({
+    seconds: new Long(date.getTime() / 1000),
+    nanos,
+  });
+}
```

Comment:
I wonder how to best move this logic to CosmJS. But I would like to avoid a `Long` dependency. I could offer `{ seconds: number, nanos: number}`, which at least move the calculation.